src/rdkafka_mock_handlers.c

```diff
@@ -1049,19 +1089,33 @@ static int rd_kafka_mock_handle_Metadata(rd_kafka_mock_connection_t *mconn,
                         const rd_kafka_topic_partition_t *rktpar =
                             &requested_topics->elems[i];
                         rd_kafka_mock_topic_t *mtopic;
+                        char *topic_name;
                         rd_kafka_resp_err_t err = RD_KAFKA_RESP_ERR_NO_ERROR;
+                        rd_kafka_Uuid_t topic_id =
+                            rd_kafka_topic_partition_get_topic_id(rktpar);
+
+                        if (rkbuf->rkbuf_reqhdr.ApiVersion >= 12 &&
+                            !RD_KAFKA_UUID_IS_ZERO(topic_id)) {
+                                mtopic = rd_kafka_mock_topic_find_by_id(
+                                    mcluster, topic_id);
+                        } else
+                                mtopic = rd_kafka_mock_topic_find(
+                                    mcluster, rktpar->topic);
+
+                        if (mtopic)
+                                topic_name = mtopic->name;
+                        else
+                                topic_name = rktpar->topic;
```

Comment:
I think if the `mtopic` is not found, we should not send incorrect topic name which was sent in the request. Lets also check what real broker is doing.

Reply:
My bad.